apps/weightgurus/weight_gurus.star

```diff
@@ -0,0 +1,489 @@
+"""
+Applet: Weight Gurus
+Summary: Displays recent weigh-ins
+Description: Displays the most recent weigh-ins from your Weight Gurus scale.
+Author: grantmd
+"""
+
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("humanize.star", "humanize")
+load("math.star", "math")
+load("render.star", "render")
+load("schema.star", "schema")
+load("time.star", "time")
+
+#App Settings
+CACHE_TTL = 60 * 60 * 24  # updates once daily
+
+#API information from https://gist.github.com/MarkWalters-pw/08ea0e8737e3e4d11f70427ef8fdc7df and https://github.com/merval/weightgurus
+WEIGHTGURUS_LOGIN = "https://api.weightgurus.com/v3/account/login"
+WEIGHTGURUS_HISTORY = "https://api.weightgurus.com/v3/operation/?%s"
+
+#Weight Gurus Data Display
+DISPLAY_FONT = "CG-pixel-3x5-mono"
+FAT_COLOR = "#b9d9eb"
+KILOGRAMS_TO_POUNDS_MULTIPLIER = float(2.205)
+WEIGHT_COLOR = "#00B0B9"
+WHITE_COLOR = "#FFF"
+
+# buildifier: disable=function-docstring
+def main(config):
+    #get user settings
+    email = config.get("email")
+    password = config.get("password")
+
+    period = config.get("period") or "0"
+    system = config.get("system") or "imperial"
+    secondary_display = config.get("second") or "none"
+
+    #setup our variables for holding the data we get back and display
+    operations = None
+
+    if not email or not password:
+        print("No email and password set, using example data")
+        operations = json.decode(EXAMPLE_DATA)
+    else:
+        #get an access token
+        access_token = cache.get("access_token")
```

Comment:
Cache is scoped to the app, but not to individual users. This will use the same cache key for all users, likely resulting in some users data being shared with all others.

Reply:
Oh, I thought the cache was per-device. So keying this off of email address is probably a good idea?